### PR TITLE
inspector: init at 0.2.0

### LINKED ISSUE #347834: Package request: inspector
**Project description**

A Gtk4 Libadwaita wrapper for various cli commands that give information about your system.

**Metadata**

* homepage URL:
* source URL: https://github.com/Nokse22/inspector
* license: gpl3
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by mksafavi**
## Things done

closes https://github.com/NixOS/nixpkgs/issues/347834

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/in/inspector/package.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  libadwaita,
+  python3Packages,
+  iproute2,
+  util-linux,
+  coreutils,
+  usbutils,
+  pciutils,
+}:
+
+#stdenv.mkDerivation rec {
```
Comment (HeitorAugustoLN) [inline pkgs/by-name/in/inspector/package.nix:20]: This probably should be removed.

File: pkgs/by-name/in/inspector/package.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  strictDeps = true;
+
+  meta = with lib; {
+    homepage = "https://github.com/Nokse22/inspector";
+    description = "A Gtk4 Libadwaita wrapper for various system info cli commands";
+    license = licenses.gpl3Plus;
+    platforms = platforms.linux;
+    mainProgram = "inspector";
+    maintainers = with maintainers; [ mksafavi ];
+  };
```
Comment (HeitorAugustoLN) [inline pkgs/by-name/in/inspector/package.nix:64]: `with lib` in meta is highly discouraged nowadays
```suggestion
  meta = {
    homepage = "https://github.com/Nokse22/inspector";
    description = "A Gtk4 Libadwaita wrapper for various system info cli commands";
    license = lib.licenses.gpl3Plus;
    platforms = lib.platforms.linux;
    mainProgram = "inspector";
    maintainers = with lib.maintainers; [ mksafavi ];
  };
```

Comment (mksafavi) [reply pkgs/by-name/in/inspector/package.nix]: I keep forgetting about this :)

File: pkgs/by-name/in/inspector/package.nix [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    usbutils
+    pciutils
+  ];
+
+  strictDeps = true;
+
+  meta = {
+    homepage = "https://github.com/Nokse22/inspector";
+    description = "A Gtk4 Libadwaita wrapper for various system info cli commands";
+    license = lib.licenses.gpl3Plus;
```
Comment (HeitorAugustoLN) [inline pkgs/by-name/in/inspector/package.nix:59]: Looks like the project has 2 licenses
https://github.com/Nokse22/inspector/blob/745e61b1720f600ebe0d21dc330eb3067ac97b5b/data/io.github.nokse22.inspector.appdata.xml.in#L4

```suggestion
    license = with lib.licenses; [
      gpl3Plus
      cc0
    ];
```

File: pkgs/by-name/in/inspector/package.nix [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    coreutils
+    usbutils
+    pciutils
+  ];
+
+  strictDeps = true;
+
+  meta = {
+    homepage = "https://github.com/Nokse22/inspector";
+    description = "A Gtk4 Libadwaita wrapper for various system info cli commands";
```
Comment (HeitorAugustoLN) [inline pkgs/by-name/in/inspector/package.nix:58]: Almost forgot about this 😅
```suggestion
    description = "Gtk4 Libadwaita wrapper for various system info cli commands";
```

**Comment by HeitorAugustoLN** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349656`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>inspector</li>
  </ul>
</details>

**Review by HeitorAugustoLN — APPROVED** [review]
Seems to be working fine.

**Review by drupol — APPROVED** [review]

**Pull request merged**